projects/components/src/dropdown/dynamic-dropdown-position.directive.ts

```diff
@@ -143,34 +141,34 @@ export class DynamicDropdownPositionDirective implements AfterContentInit {
         const contentAreaRect = this.contentAreaElement.getBoundingClientRect();
         // If the dropdown is getting clipped on the left and there is enough place on the right, shift the dropdown to right
         if (
-            dropdownMenuRect.left + MENU_BUFFER_SPACE <= contentAreaRect.left &&
-            dropdownTriggerRect.right + dropdownMenuRect.width <= contentAreaRect.right
+            this.dropdownMenuRect.left + MENU_BUFFER_SPACE <= contentAreaRect.left &&
```

Comment:
When you're using a reference this many times, you should create a local to make the code look less busy. And it would have saved you time

```const {dropdownMenuRect} = this;```